File: src/cs/Library/Elements/Slider.cs
namespace JJA.Anperi.Lib.Elements
{
    public class Slider : Element
    {
        public override string type => "slider";

        public int min { get; set; } = 0;
        public int max { get; set; } = 100;
        public int progress { get; set; } = 50;
        public int step_size { get; set; } = 2;
    }
}

File: src/java/AnperiLib/src/com/jannes_peters/anperi/lib/elements/Slider.java
package com.jannes_peters.anperi.lib.elements;

import org.json.simple.JSONObject;

public class Slider extends Element {
    public Slider(String id, int min, int max, int value, int stepSize) {
        super("slider", id);
        this.min = min;
        this.max = max;
        this.progress = value;
        this.step_size = stepSize;
    }

    int min, max, progress, step_size;

    @Override
    protected JSONObject createJSONObject() {
        JSONObject jo = super.createJSONObject();
        jo.put("min", min);
        jo.put("max", max);
        jo.put("progress", progress);
        jo.put("step_size", step_size);
        return jo;
    }

}
